### PR TITLE
UCP/TAG: tag specific initialization removed from general init function

### PR DESCRIPTION
**Pull request by hoopoepg**
- removed tag specific initializations from general request_start call
  to eliminate data conflict

### CODE REVIEW

File: src/ucp/tag/tag_send.c [hunk 1]
```diff
@@ -84,8 +84,13 @@ ucp_tag_send_req(ucp_request_t *req, size_t dt_count,
         } else {
             return UCS_STATUS_PTR(status);
         }
+    } else if ((req->send.uct.func == proto->zcopy_multi) ||
```
Comment (evgeny-leksikov) [inline src/ucp/tag/tag_send.c]: `ucs_unlikely`?

File: src/ucp/tag/tag_send.c [hunk 2]
```diff
@@ -84,8 +84,13 @@ ucp_tag_send_req(ucp_request_t *req, size_t dt_count,
...
+               (req->send.uct.func == proto->bcopy_multi)) {
+        req->send.tag.message_id  = req->send.ep->worker->tm.am.message_id++;
+        req->send.tag.am_bw_index = 1;
     }
 
+
```
Comment (evgeny-leksikov) [inline src/ucp/tag/tag_send.c]: redundant line

File: src/ucp/tag/tag_send.c (hunk 1 above)
Comment (hoopoepg) [reply src/ucp/tag/tag_send.c]: ok, added

File: src/ucp/tag/tag_send.c (hunk 2 above)
Comment (hoopoepg) [reply src/ucp/tag/tag_send.c]: yep, removed

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9749/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9756/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:bgate:retest

File: src/ucp/core/ucp_request.c [hunk 3]
```diff
@@ -309,28 +309,28 @@ ucs_status_t
 ucp_request_send_start(ucp_request_t *req, ssize_t max_short,
                        size_t zcopy_thresh, size_t zcopy_max, size_t dt_count,
                        const ucp_ep_msg_config_t* msg_config,
-                       const ucp_proto_t *proto)
+                       const ucp_proto_t *proto, int *multi_send)
```
Comment (yosefe) [inline src/ucp/core/ucp_request.c]: just noticed, we should probably move this to ucp_request.inl to make it work..

Comment (hoopoepg) [reply src/ucp/core/ucp_request.c]: but this is too big function for inline

Comment (yosefe) [reply src/ucp/core/ucp_request.c]: so let's revert to checking proto send func..

Comment (hoopoepg) [reply src/ucp/core/ucp_request.c]: ok, done

**Comment by yosefe** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9772/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9781/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9802/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9813/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9825/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:bgate:retest

**Pull request merged**